lib/wine_o_inventory_api_web/plugs/check_json_spec.ex

```diff
@@ -0,0 +1,26 @@
+defmodule WineOInventoryApiWeb.Plugs.CheckJsonSpec do
+    use WineOInventoryApiWeb, :controller
+    import Plug.Conn
+
+    def init(_params) do
+    end
+
+    def call(conn, _params) do
+        inspect_json(conn.method, conn)
+    end
+
+    def inspect_json("POST", conn) do
+        request_body = conn.body_params
+            if (request_body["data"] == nil || request_body["data"]["attributes"] == nil) do
+                conn
+                |> put_status(:bad_request)
+                |> json(%{errors: %{detail: "Bad Request"}})
```

Comment:
Just a note that JSON:API also has a specific format for error objects. https://jsonapi.org/examples/#error-objects. You can either choose to implement the spec all the way, which helps with consistency and maintainability, or take what you like from it and create your own custom way of handling things, which might be easier in the short term and allow you to develop more quickly. I'd prefer you choose whatever approach allows you to learn more.